### PR TITLE
MultiLanguageString handling

### PR DESCRIPTION
**Pull request by bwiedmann**

### CODE REVIEW

File: src/main/java/sirius/db/mixing/properties/MultiLanguageStringProperty.java [hunk 1]
```diff
@@ -206,6 +232,19 @@ protected boolean isConsideredNull(Object propertyValue) {
...
+    public void parseValues(Object entity, Values values) {
+        MultiLanguageString multiLanguageString = getMultiLanguageString(entity);
+        Map<String, String> mlsMap = new HashMap<>(multiLanguageString.data());
+        mlsMap.put(MultiLanguageString.FALLBACK_KEY, values.at(0).getString());
+        setValue(entity, mlsMap);
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    protected void checkNullability(Object propertyValue) {
```
Comment (andyHa) [inline src/main/java/sirius/db/mixing/properties/MultiLanguageStringProperty.java:245]: // Intentionally left empty as logic is implemented in onBeforeSaveChecks

-> Sonarlint should normally mark this with an inspection (leaving method empty...

File: src/main/java/sirius/db/mixing/properties/MultiLanguageStringProperty.java [hunk 2]
```diff
@@ -97,6 +99,30 @@ protected void onBeforeSaveChecks(Object entity) {
                                                    .handle();
                                });
         }
+
+        Object propertyValue = getValue(entity);
+        boolean withFallback = multiLanguageString.isWithFallback();
+        Map<String, String> values = ((Map<String, String>) propertyValue);
```
Comment (andyHa) [inline src/main/java/sirius/db/mixing/properties/MultiLanguageStringProperty.java:105]: replace with "multiLanguageString.data()" which doesn't require any cast magic...

File: src/main/java/sirius/db/mixing/properties/MultiLanguageStringProperty.java [hunk 3]
```diff
@@ -97,6 +99,30 @@ protected void onBeforeSaveChecks(Object entity) {
...
+
+        if (!isNullable()) {
```
Comment (andyHa) [inline src/main/java/sirius/db/mixing/properties/MultiLanguageStringProperty.java:107]: if (isNullable()) {
   return;
}

Comment (andyHa) [reply src/main/java/sirius/db/mixing/properties/MultiLanguageStringProperty.java:107]: can be don in line 102

File: src/main/java/sirius/db/mixing/properties/MultiLanguageStringProperty.java [hunk 4]
```diff
@@ -97,6 +99,30 @@ protected void onBeforeSaveChecks(Object entity) {
...
+            if(withFallback && Strings.isEmpty(values.get(MultiLanguageString.FALLBACK_KEY))) {
```
Comment (andyHa) [inline src/main/java/sirius/db/mixing/properties/MultiLanguageStringProperty.java:108]: please run code format before creating a PR - nothing worse than merging code style changes. and please invoke maven at least once to have proper code style setting in place.

File: src/main/java/sirius/db/mixing/properties/MultiLanguageStringProperty.java [hunk 5]
```diff
@@ -97,6 +99,30 @@ protected void onBeforeSaveChecks(Object entity) {
...
+                throw Exceptions.createHandled()
+                                .error(new InvalidFieldException(getName()))
+                                .withNLSKey("MultiLanguageStringProperty.fallbackNotSet")
+                                .set("field", getFullLabel())
+                                .handle();
+            }
+
+            boolean consideredEmpty = isConsideredNull(propertyValue);
```
Comment (andyHa) [inline src/main/java/sirius/db/mixing/properties/MultiLanguageStringProperty.java:116]: shoudln't we replace this with Strings.isEmpty as it is way clearer what happens here. Also I#d use pretty much a stream pipeline as defined above

File: src/main/java/sirius/db/mixing/properties/MultiLanguageStringProperty.java [hunk 6]
```diff
@@ -97,6 +99,30 @@ protected void onBeforeSaveChecks(Object entity) {
...
+            if (consideredEmpty) {
+                throw Exceptions.createHandled()
+                                .error(new InvalidFieldException(getName()))
+                                .withNLSKey("MultiLanguageStringProperty.empty")
+                                .set("field", getFullLabel())
+                                .handle();
+            }
+        }
+
         super.onBeforeSaveChecks(entity);
```
Comment (andyHa) [inline src/main/java/sirius/db/mixing/properties/MultiLanguageStringProperty.java:126]: can be removed, this is empty by definition

**Review by andyHa — CHANGES_REQUESTED** [review]

File: src/main/java/sirius/db/mixing/properties/MultiLanguageStringProperty.java (hunk 4 above)
Comment (bwiedmann) [reply src/main/java/sirius/db/mixing/properties/MultiLanguageStringProperty.java:108]: Thanks you the hint - i'll recheck my settings and do a mvn clean compile again to get the idea codeformat settings.

**Comment by bwiedmann** [comment]
MLS related files will be moved over to sirius-biz for smarter integration.

**Pull request closed without merging**